src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs

```diff
@@ -219,46 +220,43 @@ private IEnumerable<XRefSpec> GetXRefFromReference(PageViewModel vm)
         }
 
         private static IEnumerable<XRefSpec> GetXRefInfo(ItemViewModel item, string key,
-            List<DataContracts.Common.ReferenceViewModel> references)
+            List<ReferenceViewModel> references)
         {
             var result = new XRefSpec
             {
                 Uid = item.Uid,
                 Name = item.Name,
-                Href = key,
+                Href = ((RelativePath)key).UrlEncode().ToString(),
                 CommentId = item.CommentId,
             };
-            if (!string.IsNullOrEmpty(item.NameForCSharp))
+            if (item.Names.Count > 0)
```

Comment:
yes, but it can avoid create an enumerator for sorted list.
[https://msdn.microsoft.com/en-us/library/ms132335(v=vs.110).aspx](https://msdn.microsoft.com/en-us/library/ms132335(v=vs.110).aspx)

---
In reply to: [95104669](https://github.com/dotnet/docfx/pull/1129#discussion_r95104669) [](ancestors = 95104669)